### PR TITLE
[wip] vendor: update aws-sdk-go to v1.12.5

### PR DESCRIPTION
**Pull request by adnxn**
### Summary

updating aws-sdk-go to [v1.12.5](https://github.com/aws/aws-sdk-go/releases/tag/v1.12.5): [b489cce](https://github.com/aws/aws-sdk-go/commit/b489cce1dd1ada60f519fe0dc70c096379a42214)
### Implementation details

n/a
### Testing

- [x] Builds on Linux (`make release`)
- [x] Builds on Windows (`go build -out amazon-ecs-agent.exe ./agent`)
- [x] Unit tests on Linux (`make test`) pass
- [x] Unit tests on Windows (`go test -timeout=25s ./agent/...`) pass
- [x] Integration tests on Linux (`make run-integ-tests`) pass
- [x] Integration tests on Windows (`.\scripts\run-integ-tests.ps1`) pass
- [x] Functional tests on Linux (`make run-functional-tests`) pass
- [x] Functional tests on Windows (`.\scripts\run-functional-tests.ps1`) pass

New tests cover the changes: 
no
### Description for the changelog

n/a
### Licensing

This contribution is under the terms of the Apache 2.0 License: 
yes

### CODE REVIEW

File: agent/Godeps/Godeps.json [hunk 1]
```diff
@@ -436,6 +436,21 @@
...
 			"ImportPath": "golang.org/x/tools/imports",
 			"Rev": "d63e2b22b05a9682de336cd4802bba367ed429e7"
+		},
+		{
+			"ImportPath": "github.com/aws/aws-sdk-go/internal/shareddefaults",
+			"Comment": "v1.12.5",
+			"Rev": "b489cce1dd1ada60f519fe0dc70c096379a42214"
+		},
+		{
+			"ImportPath": "github.com/aws/aws-sdk-go/vendor/github.com/go-ini/ini",
```
[the remark this answers is not in the document]
Comment (adnxn) [reply agent/Godeps/Godeps.json]: hrm apparently it's a known issue https://github.com/tools/godep/issues/498#issuecomment-238946586. ill try the work around described there.

**Comment by adnxn** [comment]
blocked on dep migration #1009

**Comment by aaithal** [comment]
@adnxn why? The two should be independent of each other

**Comment by adnxn** [comment]
closing since aws-sdk-go was updated with the cgroup changes.

**Pull request closed without merging**